workshops/Image Sharing Site/README.md

```diff
@@ -0,0 +1,459 @@
+  const file = document.querySelector("#photo").files[0];
+  const task = ref.child(file.name).put(file);
+  task
+    .then((snapshot) => snapshot.ref.getDownloadURL())
+    .then((url) => {
+      link = url;
+      document.getElementById("status").innerHTML = "Now you can upload it.";
+    });
+}
+```
+
+In this function, we are returning a value(here the upload status) to the h1 tag(#status) to display that the image is setting up. Then we are declaring a variable ref as firebase storage. Next, we declare another variable file as the input field file fetch(i.e, the image that we select to upload) The variable task is to put the file into the firebase storage.
+Then we are downloading the URL of the image that we just uploaded to our firebase storage.
+After that, we display that the image is uploaded so you can post it. This adds the URL we just downloaded to the firebase firestore.
+Let's push it to the firestore. This happens when you click the upload image button.
+
+```js
+function upload() {
+  const name = document.getElementById("name").value;
+  if (link === null) {
+    document.getElementById("status").innerHTML = "Select an image.";
+  } else if (name.length <= 2) {
+    document.getElementById("status").innerHTML =
+      "Name should be atleast 3 charector long.";
+  } else {
+    document.getElementById("status").innerHTML = "Uploading...";
+    const url = {
+      url: link,
+      name: name,
+    };
+    db.collection("pictures")
+      .doc()
+      .set(url)
+      .then(() => {
+        location.reload();
+      });
+  }
+}
+```
+
+We get the name the user gives in the input tag earlier and check if the URL of the image is null(meaning image not selected). Then we check whether the name is 3 characters long. After all this, we will display the message 'uploading...' and send our URL + name to our firestore. Now we need to display it right? Let's write the code for that.
+
+```js
+db.collection("pictures")
+  .get()
+  .then((querySnapshot) => {
+    const data = querySnapshot.docs.map((doc) => doc.data());
+    var a = 0;
+    for (var i in data) {
+      var img = document.createElement("img");
+      img.src = data[a].url;
+      img.setAttribute("width", "600");
+      var h = document.createElement("H1");
+      var t = document.createTextNode(data[a].name);
+      h.appendChild(t);
+      h.style.color = "deeppink";
+      document.getElementById("container").appendChild(h);
+      document.getElementById("container").appendChild(img);
+      a = a + 1;
+    }
+  });
+```
+
+In these lines of code, we are fetching all URLs of our images from our firestore and storing it in an array. Then we display them one after the other. Here we need to create 2 tags one for the image and one for the heading for displaying the images and names. Then we need to make a container in which we append it.
+
+```html
+<div class="seconddiv" id="container"></div>
+```
+
+### Finally we need to add CSS to make it look better.
```

Comment:
Even though this is a smaller header, I think you should still avoid making headers entire sentences. This sentence would a great first sentence under a header called something like “Adding CSS”